causallib/time_varying/GFormula.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python3
+
+import abc
+from typing import Optional, Any, OrderedDict
+import pandas as pd
+from ..estimation.base_estimator import IndividualOutcomeEstimator
+from sklearn.base import BaseEstimator as SKLearnBaseEstimator
+from .regression_curve_fitter import RegressionCurveFitter #TODO Referenced from Survival module. Assuming we need have similar wrapper class for fitting curve
+
+
+#TODO
+class CovariateModels(OrderedDict):
+    covariate: str
+    model: IndividualOutcomeEstimator
+
+
+class GFormulaBase(IndividualOutcomeEstimator):
+    """
+        GFormula base Estimator
+    """
+    def __init__(self,
+                 outcome_model: None,
+                 treatment_model: Any,
+                 covariate_models: CovariateModels,
+                 refit_models=True):
+        """
+            outcome_model(IndividualOutcomeEstimator): A causal model that estimate on individuals level
+                                                      (e.g. Standardization).
+            treatment_model (???):
+            covariate_models (dict): {
+                        ‘x1’: Standardization(LogisticRegression()),
+                        ‘x2’: Standardization(LinearRegression()),
+                            so forth
+                        },
+            refit_models (bool): if True, re-fit a the treatment model and covariate models.
+
+        """
+        super(GFormulaBase, self).__init__(lambda **x: None)
+        self.outcome_model = outcome_model
+        self.treatment_model = self._init_regressor(treatment_model)
+        self.covariate_models = covariate_models #TODO call _init_regressor() for all covariate_models
+        self.refit_models = refit_models
+
+    def _init_regressor(self, model):
+        """method to check and wrap the default learner class with RegressionCurveFitter."""
+        if isinstance(model, SKLearnBaseEstimator):
+            # Construct default curve fitter, parametric with a scikit-learn estimator
+            return RegressionCurveFitter(model)
+        else:
+            # Initialized lifelines RegressionFitter (or any implementation with a compatible API)
+            return model
+
+    @abc.abstractmethod
+    def fit(self,
+            X: pd.DataFrame,
+            a: pd.Series,
```

Comment:
@ehudkr G-computation can support multiple treatment variables. Do you think we can support a dataframe for `a`?

Reply:
Good point. I suggest we start with a single treatment at the moment, and expand it later?
(also, you mean concurrent treatment I guess, because otherwise a vector `a` can just take non-binary values, no?)

Reply:
yes i was thinking of back to back or combination of treatments. e.g. vasopressor and fluid (both could be binary)
- we can certainly start with a single one and then expand

Reply:
Comment added for single treatment variable